### PR TITLE
Miscellaneous VOXL2 board updates

### PR DESCRIPTION
**Pull request by katzfey**
### Solved Problem
Default base directory in the file system for PX4 files is different for VOXL2. Also, added ability in Qurt platform for log messages to go to the Linux side for display and logging.

### CODE REVIEW

File: platforms/common/include/px4_platform_common/defines.h [hunk 1]
```diff
@@ -95,7 +95,11 @@ __BEGIN_DECLS
 extern long PX4_TICKS_PER_SEC;
 __END_DECLS
 
+#if defined(CONFIG_ARCH_BOARD_MODALAI_VOXL2)
```
Comment (dagar) [inline platforms/common/include/px4_platform_common/defines.h:98]: We should avoid board specifics like this slipping into common headers, can we get this set another way either explicitly configured (kconfig) or derived from the CWD or something?

Comment (katzfey) [reply platforms/common/include/px4_platform_common/defines.h:98]: I'll see if I can find a better way.

Comment (katzfey) [reply platforms/common/include/px4_platform_common/defines.h]: I added a Kconfig for Posix platforms to configure it.

Comment (katzfey) [reply platforms/common/include/px4_platform_common/defines.h]: But now a bunch of checks are failing. Maybe due to the include of px4_platform_common/px4_config.h in defines.h? But seems like a real error. A file is using a variable _K but there is a macro with that same name.

Comment (katzfey) [reply platforms/common/include/px4_platform_common/defines.h]: Okay, looks like I fixed the failing test cases by being more accurate in my header file inclusion.

**Review by dagar — APPROVED** [review]

**Pull request merged**